src/workflow/runtime/WorkflowRuntime.ts

```diff
@@ -33,19 +34,19 @@ export default class WorkflowRuntime {
    * @param {WorkflowClientOptions | undefined} options - Additional options for configuring WorkflowRuntime.
    */
   constructor(options: Partial<WorkflowClientOptions> = {}) {
-    const hostAddress = this.generateHostAddress(options);
-    options.daprApiToken = this.generateDaprApiToken(options);
-    this.worker = this.buildInnerWorker(hostAddress, options);
+    const grpcEndpoint = this.generateEndpoint(options);
+    options.daprApiToken = this.getDaprApiToken(options);
+    this.worker = this.buildInnerWorker(grpcEndpoint.endpoint, options);
   }
 
-  private generateHostAddress(options: Partial<WorkflowClientOptions>): string {
-    const host = options?.clientHost ?? Settings.getDefaultHost();
-    const port = options?.clientPort ?? Settings.getDefaultGrpcPort();
-    const hostAddress = `${host}:${port}`;
-    return hostAddress;
+  private generateEndpoint(options: Partial<WorkflowClientOptions>): GrpcEndpoint {
```

Comment:
This seems to be duplicated, can we move it to a common place? Both endpoint and Api token functions.

Reply:
combined please check!